src/averageAges.js

```diff
@@ -55,7 +74,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const selectedPeople = onlyWithSon
+    ? people.filter(p => p.sex === 'm')
+    : people;
+
+  let count = 0;
+  let mother;
+
+  const total = selectedPeople.reduce((a, b) => {
```

Comment:
let's think of a better name for this variable

Reply:
and for the parameters as well

Reply:
This is because I wrote all these codes in a single night. Also been learning since then